packages/graph/hash_graph/bench/benches/representative_read/seed.rs

```diff
@@ -0,0 +1,247 @@
+
+    eprintln!(
+        "Finished seeding database {} with {} entities after {:#?}",
+        store_wrapper.bench_db_name,
+        total_entities,
+        now.elapsed().unwrap()
+    );
+}
+
+/// DOC - TODO
+pub struct Samples {
+    pub entities: HashMap<AccountId, HashMap<VersionedUri, Vec<EntityId>>>,
+}
+
+async fn get_samples(account_id: AccountId, store_wrapper: &mut StoreWrapper) -> Samples {
+    let mut samples = Samples {
+        entities: HashMap::from([(account_id, HashMap::new())]),
+    };
+
+    let sample_map = samples.entities.get_mut(&account_id).unwrap();
+
+    for entity_type_id in SEED_ENTITIES.map(|(entity_type_str, ..)| {
+        EntityType::from_str(entity_type_str)
+            .expect("could not parse entity type")
+            .id()
+            .clone()
+    }) {
+        // For now we'll just pick a sample of 50 entities.
+        let sample_entity_ids = store_wrapper
+            .store
+            .as_client()
+            .query(
+                r#"
+                -- Very naive and slow sampling, we can replace when this becomes a bottleneck
+                SELECT entity_id FROM entities
+                INNER JOIN type_ids
+                ON type_ids.version_id = entities.entity_type_version_id
+                WHERE type_ids.base_uri = $1 AND type_ids.version = $2
+                ORDER BY RANDOM()
+                LIMIT 50
+                "#,
+                &[
+                    &entity_type_id.base_uri().as_str(),
+                    &(entity_type_id.version() as i64),
+                ],
+            )
+            .await
+            .unwrap_or_else(|err| {
+                panic!("failed to sample entities for entity type `{entity_type_id}`: {err}");
+            })
+            .into_iter()
+            .map(|row| row.get(0));
+
+        match sample_map.entry(entity_type_id) {
+            Entry::Occupied(mut entry_slot) => entry_slot.get_mut().extend(sample_entity_ids),
+            Entry::Vacant(entry_slot) => {
+                entry_slot.insert(sample_entity_ids.collect::<Vec<_>>());
+            }
+        }
+    }
+
+    samples
+}
+
+pub async fn setup_and_extract_samples(store_wrapper: &mut StoreWrapper) -> Samples {
+    // TODO: We'll want to test distribution across accounts
+    //  https://app.asana.com/0/1200211978612931/1203071961523000/f
+    // We use a hard-coded UUID to keep it consistent across tests so that we can use it as a
+    // parameter argument to criterion and get comparison analysis
+    let account_id =
+        AccountId::new(Uuid::from_str("d4e16033-c281-4cde-aa35-9085bf2e7579").unwrap());
+
+    let not_empty: bool = store_wrapper
+        .store
+        .as_client()
+        .query_one(
+            r#"
+            SELECT EXISTS(SELECT 1 FROM accounts WHERE account_id=$1)
+            "#,
+            &[&account_id],
+        )
+        .await
+        .expect("failed to check if account id exists")
+        .get(0);
+
+    if !(not_empty) {
```

Comment:
```suggestion
    if !account_id_exists {
```

Reply:
Chose something similar, and added a comment